CONTRIBUTING.md

```diff
@@ -1,100 +1,239 @@
+✅ **Yay:** _(Applying this patch will) make number of textarea rows configurable_
 
-## Publishing a hotfix release from `current`
+❌ **Nay:** _(Applying this patch will) added configuration for textarea rows_
 
-If we need to publish a hotfix, a patch release should be done by running `npm run publish` from the project root, selecting "Patch" from the menu. Remember to rebase `next` against `current` after the release is completed.
+We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), a popular standard for commit messages in the open-source world. This provides a consistent and readable commit style that can also be used for automated tasks.
 
-## Publishing _canary_ versions from feature branches
+In essence, conventional commits has the following format:
+
+```
+fix(form-builder): remove empty objects from array when closing edit dialog
+```
+
+We highly recommend you use more than one line to describe your changes, if possible. This allows you to provide much more context than in the one-line message, and can provide additional terms to hit when using `git log` or GitHubs code search. Here is an example:
+
+```
+fix(form-builder): remove empty objects from array when closing edit dialog
+
+In the future we will investigate whether or not we can prevent empty objects
+from being added to the arrays in the first place. Currently this is done to
+simplify how patches are generated, as well as to ensure "placeholders" appear
+in real-time.
+```
 
-This can be done at any time by anyone and is done by `npm run publish-canary`. This will publish with the commit hash from HEAD, and can be installed with `sanity upgrade --tag=canary --save-exact`
+In more formal terms, this is how the specification defines it:
 
-# How to file an issue
+```
+<type>[optional scope]: <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+In addition to the standard `fix` and `feat` commit types, we also use the additional types defined under "Commit types" below, which is based on the [Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines).
+
+### New features
+
+Please note that if you would like to develop a new feature for the Sanity Studio, it's unlikely that we will be able to accept your contribution. However, we highly recommend that you develop and publish your new feature as a plugin. Plugins can be [parts](https://www.sanity.io/docs/extending/parts), [custom input components](https://www.sanity.io/docs/extending/custom-input-widgets), [dashboard widgets](https://www.sanity.io/docs/dashboard/creating-your-own-widget), and [Studio themes](https://www.sanity.io/docs/guides/how-to-brand-your-studio) that you and others can use to customize and extend the functionality of the Studio.
+
+### Pull Requests
```

Comment:
Should we say something about adding tests should be semi-required? Tests aren't setup everywhere yet, so maybe it would be a little harsh to require it. At the same time, adding them would be highly appreciated. 👍